spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -434,8 +434,14 @@ public Offset latestOffset(Offset startOffset, ReadLimit limit) {
 
       // if everything was OK and we consumed complete snapshot then move to next snapshot
       if (shouldContinueReading) {
+        Snapshot nextValid = nextValidSnapshot(curSnapshot);
+        if (nextValid == null) {
+          // nextValid implies all the remaining snapshots should be skipped.
+          break;
```

Comment:
In the original, there was a
```
          shouldContinueReading = false;
```
before the `break`, but that was removed as unnecessary by https://github.com/apache/iceberg/commit/afda8be25652d44d9339a79c6797b6bf20c55bd6.